src/main/java/gg/solarmc/clans/events/StatsEvent.java

```diff
@@ -0,0 +1,57 @@
+package gg.solarmc.clans.events;
+
+import gg.solarmc.loader.clans.Clan;
+import gg.solarmc.loader.clans.ClansKey;
+import org.bukkit.Server;
+import org.bukkit.attribute.Attribute;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.entity.EntityRegainHealthEvent;
+import org.bukkit.event.entity.PlayerDeathEvent;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class StatsEvent implements Listener {
+    private final Map<UUID, UUID> assists = new HashMap<>();
+
+    @EventHandler
+    public void onPlayerDamage(EntityDamageByEntityEvent event) {
+        if (event.getEntity() instanceof Player damaged && event.getDamager() instanceof Player damager)
+            assists.put(damaged.getUniqueId(), damager.getUniqueId());
```

Comment:
The way you've made this, there can be only 1 assistant per kill. `put` replaces a previous value, if one exists, at the given key